### PR TITLE
Added Support for SQL Server INSERT INTO Parsing

### LINKED ISSUE #29194: Support parsing SQL Server INSERT INTO sql
# Background
Hi community.
This issue is for #29149.

The ShardingSphere SQL parser engine helps users to parse SQL to create the AST (Abstract Syntax Tree) and visit the AST to get SQLStatement (Java Object). Currently, we are planning to enhance the support for `SQL Server` SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

# Task
This issue is to support more `SQL Server` sql parsing, as follows:
```sql
INSERT INTO #MyTempTable
VALUES (1)
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16)

```sql
INSERT INTO #t VALUES (2)
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16)

```sql
SELECT Test2Col = x FROM #t
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16)

```sql
INSERT INTO #t VALUES (1)
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16)

```sql
SELECT Test1Col = x FROM #t
```
[link](https://learn.microsoft.com/zh-cn/sql/t-sql/statements/create-table-transact-sql?view=sql-server-ver16)

# Process
1. First confirm that this is a correct `SQL Server` sql syntax, if not please leave a message under the issue and ignore it;
2. Compare SQL definitions in Official SQL Doc and ShardingSphere SQL Doc;
3. If there is any difference in ShardingSphere SQL Doc, please correct them by referring to the Official SQL Doc;
4. Run mvn install the current_file_module;
5. Check whether there are any exceptions. If indeed, please fix them. (Especially xxxVisitor.class);
6. Add new corresponding SQL case in SQL Cases and expected parsed result in Expected Statement XML;
7. Run SQLParserParameterizedTest to make sure no exceptions.

# Relevant Skills
1. Master JAVA language
2. Have a basic understanding of Antlr `g4` file
3. Be familiar with `SQL Server` SQLs

### PR DESCRIPTION
**Pull request by BykaWF**
Fixes #29194.

Changes proposed in this pull request:
It adds support for parsing the following SQL Server INSERT INTO statements:

    INSERT INTO #MyTempTable VALUES (1)
    INSERT INTO #t VALUES (2)
    SELECT Test2Col = x FROM #t
    INSERT INTO #t VALUES (1)
    SELECT Test1Col = x FROM #t

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: parser/sql/dialect/sqlserver/src/main/antlr4/imports/sqlserver/BaseRule.g4 [hunk 1]
```diff
@@ -154,6 +154,10 @@ tableName
     : (owner DOT_)? name
     ;
 
+tempTableName
```
Comment (strongduanmu) [inline parser/sql/dialect/sqlserver/src/main/antlr4/imports/sqlserver/BaseRule.g4:157]: What is the different between tableName and tempTableName?

File: test/it/parser/src/main/resources/case/dml/insert.xml [hunk 2]
```diff
@@ -2900,4 +2900,36 @@
             </from>
         </select>
     </insert>
+    <insert sql-case-id="insert_into_my_temp_table_1">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/insert.xml:2903]: Where is the orginal sql?

File: test/it/parser/src/main/resources/case/dml/insert.xml [hunk 3]
```diff
@@ -2900,4 +2900,36 @@
...
+        <values>
+            <value>
+                <assignment-value>
+                    <literal-expression value="1" start-index="31" stop-index="31" />
+                </assignment-value>
+            </value>
+        </values>
+    </insert>
+
+    <insert sql-case-id="insert_into_temp_table_2">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/insert.xml:2914]: Please rename sql-case-id to more meaningful name.

File: test/it/parser/src/main/resources/case/dml/select-expression.xml [hunk 4]
```diff
@@ -3125,4 +3125,51 @@
...
+                        <operator>=</operator>
+                        <right>
+                            <column name="x" start-index="17" stop-index="18"/>
+                        </right>
+                    </binary-operation-expression>
+                </expr>
+            </expression-projection>
+        </projections>
+        <from>
+            <simple-table name="#t" start-index="23" stop-index="25">
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/case/dml/select-expression.xml:3146]: Please use `<simple-table><simple-table />` if it doesn't have child node.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by BykaWF** [comment]
I've encountered an issue during testing where I consistently receive the following error:

**Comment by BykaWF** [comment]
I've encountered an issue during testing where I consistently receive the following error:

      org.apache.shardingsphere.sql.parser.exception.SQLParsingException: You have an error in your SQL syntax: SELECT Test1Col = x FROM #t, no viable alternative at input '<EOF>' at line 1, position 27, near [@6,27:26='<EOF>',<-1>,1:27]

Could someone provide a hint or suggestions on what might be causing this issue? I think parser doesn't get it what is `'#'`, but I have tried add '#' to IDENTIFIER_  to tableName.

**Comment by BykaWF** [comment]
@TherChenYang  thanks for helping

**Comment by strongduanmu** [comment]
Hi @BykaWF, can you solve code conflict?

**Comment by BykaWF** [comment]
I've addressed the issues pointed out in the previous review. Could you please review this updated version? Thank you!

**Comment by strongduanmu** [comment]
Hi @BykaWF, can you solve the latest code conflict?

**Comment by BykaWF** [comment]
Hi @strongduanmu, yes I can

**Comment by strongduanmu** [comment]
Since this pr has long time no response, I will close it. You can submit new pr when you are free.

**Pull request closed without merging**